paf-mvp-core-js/src/monitoring/tracer-factory.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  trace,
+  Tracer,
+  Span,
+  SpanOptions,
+  SpanContext,
+  SpanStatus,
+  Attributes as SpanAttributes,
+  AttributeValue as SpanAttributeValue,
+  TimeInput,
+  Context,
+  Exception,
+} from '@opentelemetry/api';
+
+class NoopSpanContext implements SpanContext {
+  spanId: string;
+  traceFlags: number;
+  traceId: string;
+}
+
+class NoopSpan implements Span {
+  private _spanContext: SpanContext = new NoopSpanContext();
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  addEvent(name: string, attributesOrStartTime?: SpanAttributes | TimeInput, startTime?: TimeInput): this {
+    return this;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function,@typescript-eslint/no-unused-vars
+  end(endTime?: TimeInput): void {}
+  isRecording(): boolean {
+    return false;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function,@typescript-eslint/no-unused-vars
+  recordException(exception: Exception, time?: TimeInput): void {}
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  setAttribute(key: string, value: SpanAttributeValue): this {
+    return this;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  setAttributes(attributes: SpanAttributes): this {
+    return this;
+  }
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  setStatus(status: SpanStatus): this {
+    return this;
+  }
+  spanContext(): SpanContext {
+    return this._spanContext;
```

Comment:
I think you can replace with:
```typescript
spanContext(): SpanContext {
  return {
    spanId: '',
    traceFlags: 0,
    traceId: ''
  }
}
```
And remove `this._spanContext` and `NoopSpanContext` class.
This is because in typescript, interfaces can be instanciated, without the need for a class.

Reply:
Done.